graph/generator.js

```diff
@@ -14,120 +16,83 @@ function transformArgument(arg) {
 
 function transformField(field) {
   return {
+    type: field.baseType.name,
+    kind: field.baseType.kind,
     fieldName: field.name,
-    isList: hasListType(field.type)
+    isList: hasListType(field.type),
+    args: (field.args || []).map(transformArgument)
   };
 }
 
 function objectifyField(acc, field) {
-  const descriptor = {};
-
-  Object.keys(field).filter(key => {
+  const descriptor = Object.keys(field).filter(key => {
     return (key !== 'fieldName');
-  }).forEach(key => {
-    descriptor[key] = field[key];
-  });
+  }).reduce((descriptorAcc, key) => {
+    descriptorAcc[key] = field[key];
+
+    return descriptorAcc;
+  }, {});
 
   acc[field.fieldName] = descriptor;
 
   return acc;
 }
```

Comment:
I really hate using merging enumeration operations. Reduce _isn't_ filter, and shouldn't be filtering. Filtering isn't technically a fold, where reduce is. Reduce also shouldn't mutate the input object, which is why I didn't do the delete.